### PR TITLE
Moving common health checks to the library

### PR DESCRIPTION
**Pull request by alexturkin**
### Problem
there are health checks that are common to every collector, and now it's just copied everywhere

### Solution
* DRY it!
* List with custom health checks is added to the init parameters of collector

### CODE REVIEW

File: health_checks.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                stackStatus === 'REVIEW_IN_PROGRESS') {
+                return callback(null);
+            } else {
+                return callback(errorMsg('AWS00002', 'CF stack has wrong status: ' + stackStatus));
+            }
+        }
+    });
+}
+
+function get_health_check(event, context, customChecks, callback) {
```
Comment (kkuzmin) [inline health_checks.js]: Please use camel case for function names.

File: health_checks.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+const AWS = require('aws-sdk');
+const async = require('async');
+
+function checkCloudFormationStatus(event, context, callback) {
+    var stackName = event.StackName;
+    var cloudformation = new AWS.CloudFormation();
+    cloudformation.describeStacks({StackName: stackName}, function(err, data) {
+        if (err) {
+            return callback(errorMsg('AWS00001', stringify(err)));
```
Comment (kkuzmin) [inline health_checks.js]: I'd sugget `ALAWS00001`/`ALC00001` instead of `AWS00001`

File: al_aws_collector.js [hunk 3]
```diff
@@ -73,7 +74,7 @@ class AlAwsCollector {
         })
     }
     
-    constructor(context, collectorType, ingestType, version, aimsCreds, formatFun) {
+    constructor(context, collectorType, ingestType, version, aimsCreds, formatFun, healthCheckFuns) {
```
Comment (kkuzmin) [inline al_aws_collector.js]: May I ask you please to fix class declaration header here https://github.com/alertlogic/al-aws-collector-js/pull/7/files#diff-721ff284e807cc07d540aa9120b6277eR48

File: health_checks.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            }
+        }
+    });
+}
+
+function get_health_check(event, context, customChecks, callback) {
+    customChecksFuns = customChecks.map(f => function(asyncCallback) {
+        f(event, context, asyncCallback)
+    });
+    async.waterfall([
```
Comment (kkuzmin) [inline health_checks.js]: It will be faster to run all healthcheck in parallel. Please use `async.map` here.

File: health_checks.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        }
+    });
+}
+
+function getHealthStatus(event, context, customChecks, callback) {
+    customChecksFuns = customChecks.map(f => function(asyncCallback) {
+        f(event, context, asyncCallback)
+    });
+    async.parallel([
```
Comment (alexturkin) [inline health_checks.js]: @kkuzmin we discussed `async.map` but I think `async.parallel` suits us better: 

https://caolan.github.io/async/docs.html#parallel

* Run the tasks collection of functions in parallel, without waiting until the previous function has completed. 
* If any of the functions pass an error to its callback, the main callback is immediately called with the value of the error. 

So it's basically the same behaviour as `waterfall`, so we don't even have to change interfaces

Comment (alexturkin) [reply health_checks.js]: @kkuzmin are you OK with `async.parallel`?

File: health_checks.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+const async = require('async');
+
+/**
+ * @function
+ *
+ * @param {Object} event - checkin event for Lambda function, can be used to send needed parameters for health checks. Example: stackName.
+ * @param {Object} context - context of Lambda's function.
+ * @param {function} callback - callback, which is called by health check when it's done.
+ *
+ * @returns {(null|ErrorMsg)} returns null if all OK or ErrorMsg if something was wrong
```
Comment (alexturkin) [inline health_checks.js]: @kkuzmin not sure if it's really correct way to put it with `@returns`, as actually `callback` is called on return... do you have an idea how it's usually  being done with jsDoc?

Comment (kkuzmin) [reply health_checks.js]: Good example is here https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/keyVault/lib/keyVault.js#L44-L56

**Pull request merged**